honeybee_energy/simulation/control.py

```diff
@@ -0,0 +1,206 @@
+                 do_plant_sizing=True, run_for_sizing_periods=False,
+                 run_for_run_periods=True):
+        """Initialize SimulationControl.
+
+        Args:
+            do_zone_sizing: Boolean for whether the zone sizing calculation
+                should be run. Default: True.
+            do_system_sizing: Boolean for whether the system sizing calculation
+                should be run. Default: True.
+            do_plant_sizing: Boolean for whether the plant sizing calculation
+                should be run. Default: True.
+            run_for_sizing_periods: Boolean for whether the simulation should
+                be run for the sizing periods.
+            run_for_run_periods: Boolean for whether the simulation should
+                be run for the run periods.
+        """
+        self.do_zone_sizing = do_zone_sizing
+        self.do_system_sizing = do_system_sizing
+        self.do_plant_sizing = do_plant_sizing
+        self.run_for_sizing_periods = run_for_sizing_periods
+        self.run_for_run_periods = run_for_run_periods
+
+    @property
+    def do_zone_sizing(self):
+        """Get or set a boolean for whether the zone sizing calculation is run."""
+        return self._do_zone_sizing
+
+    @do_zone_sizing.setter
+    def do_zone_sizing(self, value):
+        self._do_zone_sizing = bool(value)
+
+    @property
+    def do_system_sizing(self):
+        """Get or set a boolean for whether the system sizing calculation is run."""
+        return self._do_system_sizing
+
+    @do_system_sizing.setter
+    def do_system_sizing(self, value):
+        self._do_system_sizing = bool(value)
+
+    @property
+    def do_plant_sizing(self):
+        """Get or set a boolean for whether the plant sizing calculation is run."""
+        return self._do_plant_sizing
+
+    @do_plant_sizing.setter
+    def do_plant_sizing(self, value):
+        self._do_plant_sizing = bool(value)
+
+    @property
+    def run_for_sizing_periods(self):
+        """Get or set a boolean for whether the simulation is run for sizing periods."""
+        return self._run_for_sizing_periods
+
+    @run_for_sizing_periods.setter
+    def run_for_sizing_periods(self, value):
+        self._run_for_sizing_periods = bool(value)
+
+    @property
+    def run_for_run_periods(self):
+        """Get or set a boolean for whether the simulation is run for run periods."""
+        return self._run_for_run_periods
+
+    @run_for_run_periods.setter
+    def run_for_run_periods(self, value):
+        self._run_for_run_periods = bool(value)
+
+    @classmethod
+    def from_idf(cls, idf_string):
+        """Create a SimulationControl object from an EnergyPlus IDF text string.
+
+        Args:
+            idf_string: A text string fully describing an EnergyPlus
+                SimulationControl definition.
+        """
+        # check the inputs
+        ep_strs = parse_idf_string(idf_string, 'SimulationControl,')
```

Comment:
No it's not needed. I think it's just a good habit since many E+ object names are concatenations of other object names with more words. Putting the comma there always ensures that I have the correct type of IDF string.